src/bindings/python/src/openvino/runtime/ie_api.py

```diff
@@ -223,9 +223,10 @@ class CompiledModel(CompiledModelBase):
     multiple optimization transformations, then mapping to compute kernels.
     """
 
-    def __init__(self, other: CompiledModelBase) -> None:
+    def __init__(self, other: CompiledModelBase, weights: Optional[bytes] = None) -> None:
         # Private memeber to store already created InferRequest
         self._infer_request: Optional[InferRequest] = None
+        self._weights = weights
```

Comment:
does it mean weights are stored even when we compiled model and don't really need it?

Reply:
No, if they are not passed in constructor, these are set to `None`. So nothing is stored there.

Reply:
yes, but for case when they are passed, we store them even plugin copied them internally ?

Reply:
@ilya-lavrenov I thought that the question https://github.com/openvinotoolkit/openvino/pull/23571#discussion_r1533390155 was about keeping the weights alive as Tensor is being used to share them (void* overload). Which basically means that weights should live together with CompiledModel.

What is the case that plugin copies them? How can one detect it from API level? If there is no way to detect it on API level, I think this is something to improve on.

Reply:
@ilya-lavrenov, @jiwaszki, thanks for your feedback. 
I think we can consider plugin copy weights case in a separate task.